msg/WorldBody.msg

```
@@ -0,0 +1,45 @@
+# Representation of a single body into the world 
+
+# Possible constants to fill into the type field
+uint8 UNDEFINED=0
+uint8 PRIMITIVE_BODY=1
+uint8 MESH_BODY=2
+uint8 URDF_BODY=3
+
+# The type of body that is represented
+uint8 type
+
+# The name with which the body shall be addressed in the world
+# Note: needs to be unique. If the name is already taken, an error will be returned
+string name
+
+# Pose of the body in the world. TF will be used to resolve the header information.
+geometry_msgs/PoseStamped pose
+
+# Flag that specifies whether this body is rigidly attached to an existing link
+bool rigidly_attached
```

Not sure what is better design, but we could also interpret parent_body_name == '' or an invalid parent_body_name as not rigidly attached.

---

It is possible to interpret missing or incorrect names like that. However, I'd rather report an error in such a case. Hence, I vote to keep the bool.